core/src/main/java/org/polypheny/db/sql/ddl/altermaterializedview/SqlAlterMaterializedAddIndex.java

```diff
@@ -0,0 +1,166 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.sql.ddl.altermaterializedview;
+
+import static org.polypheny.db.util.Static.RESOURCE;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.polypheny.db.adapter.DataStore;
+import org.polypheny.db.catalog.Catalog.TableType;
+import org.polypheny.db.catalog.entity.CatalogTable;
+import org.polypheny.db.catalog.exceptions.GenericCatalogException;
+import org.polypheny.db.catalog.exceptions.UnknownColumnException;
+import org.polypheny.db.catalog.exceptions.UnknownDatabaseException;
+import org.polypheny.db.catalog.exceptions.UnknownKeyException;
+import org.polypheny.db.catalog.exceptions.UnknownSchemaException;
+import org.polypheny.db.catalog.exceptions.UnknownTableException;
+import org.polypheny.db.catalog.exceptions.UnknownUserException;
+import org.polypheny.db.ddl.DdlManager;
+import org.polypheny.db.ddl.exception.AlterSourceException;
+import org.polypheny.db.ddl.exception.IndexExistsException;
+import org.polypheny.db.ddl.exception.MissingColumnPlacementException;
+import org.polypheny.db.ddl.exception.UnknownIndexMethodException;
+import org.polypheny.db.jdbc.Context;
+import org.polypheny.db.sql.SqlIdentifier;
+import org.polypheny.db.sql.SqlNode;
+import org.polypheny.db.sql.SqlNodeList;
+import org.polypheny.db.sql.SqlUtil;
+import org.polypheny.db.sql.SqlWriter;
+import org.polypheny.db.sql.ddl.SqlAlterMaterializedView;
+import org.polypheny.db.sql.parser.SqlParserPos;
+import org.polypheny.db.transaction.Statement;
+import org.polypheny.db.transaction.TransactionException;
+import org.polypheny.db.util.ImmutableNullableList;
+
+public class SqlAlterMaterializedAddIndex extends SqlAlterMaterializedView {
+
+
+    private final SqlIdentifier table;
+    private final SqlIdentifier indexName;
+    private final SqlIdentifier indexMethod;
+    private final SqlNodeList columnList;
+    private final boolean unique;
+    private final SqlIdentifier storeName;
+
+
+    public SqlAlterMaterializedAddIndex(
+            SqlParserPos pos,
+            SqlIdentifier table,
+            SqlNodeList columnList,
+            boolean unique,
+            SqlIdentifier indexMethod,
+            SqlIdentifier indexName,
+            SqlIdentifier storeName ) {
+        super( pos );
+        this.table = Objects.requireNonNull( table );
+        this.columnList = Objects.requireNonNull( columnList );
+        this.unique = unique;
+        this.indexName = indexName;
+        this.indexMethod = indexMethod;
+        this.storeName = storeName;
+    }
+
+
+    @Override
+    public List<SqlNode> getOperandList() {
+        return ImmutableNullableList.of( table, columnList, storeName, indexMethod, indexName );
+    }
+
+
+    @Override
+    public void unparse( SqlWriter writer, int leftPrec, int rightPrec ) {
```

Comment:
This is not correct

Reply:
Corrected unparse method, it now unparses "alter materialized view".